core/src/main/java/org/mule/runtime/core/internal/metadata/cache/DefaultPersistentMetadataCacheManager.java

```diff
@@ -139,6 +140,7 @@ public void dispose(String keyHash) {
         if (isBlank(keyHash)) {
           clearMetadataCaches();
         } else {
+          LOGGER.debug(format("Removing cache in OS with ID '%s'", key));
```

Comment:
Check if debug level is enabled in the logger before logging.  Also, is string formatting necessary or does the .debug method accept extra arguments for formatting?

Reply:
Sure, I forgot to check if it debug severity is enabled. I would do the changes.